applications/FluidDynamicsApplication/custom_conditions/navier_stokes_wall_condition.cpp

```diff
@@ -633,6 +644,133 @@ void NavierStokesWallCondition<TDim,TNumNodes>::ComputeGaussPointBehrSlipRHSCont
     KRATOS_CATCH("")
 }
 
+
+
+template<unsigned int TDim, unsigned int TNumNodes>
+void NavierStokesWallCondition<TDim,TNumNodes>::ComputeGaussPointNavierSlipRHSContribution( VectorType& rRightHandSideVector,
+                                                                                            const ConditionDataStruct& rDataStruct )
+{
+    KRATOS_TRY
+
+    GeometryType& rGeom = this->GetGeometry();
+
+    array_1d<double,TNumNodes> nodal_beta = ZeroVector(TNumNodes);
+	for (unsigned int nnode=0; nnode < TNumNodes; nnode++){
+        const double viscosity = rGeom[nnode].GetSolutionStepValue(DYNAMIC_VISCOSITY);
+        const double navier_slip_length = rGeom[nnode].GetValue(SLIP_LENGTH);
+        KRATOS_ERROR_IF_NOT( navier_slip_length > 0.0 ) << "Negative or zero slip length was defined" << std::endl;
+
+        nodal_beta[nnode] = viscosity / navier_slip_length;
+	}
+
+    // Retrieve the nodal consistent normal vectors, normalize, and store them for each node
+    std::vector<array_1d<double,3>> NodalNormals(TNumNodes);
+	for (unsigned int nnode=0; nnode < TNumNodes; nnode++){
+
+        NodalNormals[nnode] = rGeom[nnode].FastGetSolutionStepValue(NORMAL);
+        double sumOfSquares = 0.0;
+        for (unsigned int j = 0; j < 3; j++){
+            sumOfSquares += NodalNormals[nnode][j] * NodalNormals[nnode][j];
+        }
+        NodalNormals[nnode] = ( 1.0 / sqrt(sumOfSquares) ) *  NodalNormals[nnode];
```

Comment:
Did you check that what is saved in `NORMAL` is not already normalized? (just wondering, I'm not really sure). If it is not already normalized, you can do this inside the loop in where you compute the Gauss pt. contribution. In this manner you avoid an extra loop as well as the allocation of the `NordalNormals` vector.

Reply:
No, it is not normalized yet (I re-checked in the cpp-test). The respective block was moved.